command/agent/dns_test.go

```diff
@@ -4377,16 +4418,16 @@ func TestDNS_Compression_trimUDPResponse(t *testing.T) {
 	t.Parallel()
 	config := &DefaultConfig().DNSConfig
 
-	m := dns.Msg{}
-	trimUDPResponse(config, &m)
+	req, m := dns.Msg{}, dns.Msg{}
```

Comment:
Can you add another trim case? This has been a notorious source of subtle bugs so is worth poking a bit.